libmscore/layout.cpp

```diff
@@ -792,12 +794,22 @@ static QPair<qreal, qreal> layoutAccidental(AcEl* me, AcEl* above, AcEl* below,
       for (int i = 0; i < lns; ++i) {
             Note* ln = leftNotes[i];
             int lnLine = ln->line();
-            qreal lnLedgerAdjust = 0.0;
-            if (lnLine <= -2 || lnLine >= staff->lines(tick) * 2)
-                  lnLedgerAdjust = ledgerAdjust;
             qreal lnTop = (lnLine - 1) * 0.5 * sp;
             qreal lnBottom = lnTop + sp;
             if (me->top - lnBottom <= pnd && lnTop - me->bottom <= pnd) {
+                  qreal lnLedgerAdjust = 0.0;
+                  if (lnLine <= -2 || lnLine >= staff->lines(tick) * 2) {
+                        // left note has a ledger line we probably need to clear horizontally as well
+                        // except for accidentals that clear the last extended ledger line vertically
+                        // in these cases, the accidental may tuck closer
+                        Note* lastLnNote = lnLine < 0  ? leftNotes[0] : leftNotes[lns - 1];
+                        int lastLnLine = lastLnNote->line();
+                        qreal ledgerY = (lastLnLine / 2) * sp;
```

Comment:
Here is where I still lnLedgerAdjust, but only if the accidental does not in fact vertically clear the last extended ledger line.  That allows this case to tuck the accidental over the ledger line, as shown in the topmost accidental of the the second and third chords here:

![image](https://user-images.githubusercontent.com/1799936/101863938-0e0b7400-3b31-11eb-95f8-f683f3b02ea2.png)